### PR TITLE
✨  Support the use of custom api through an object as a response

### PR DESCRIPTION
**Pull request by Kvindee**

### CODE REVIEW

File: src/main/java/com/mindee/CLI.java [hunk 1]
```diff
@@ -101,10 +90,32 @@ void passportMethod() throws IOException {
...
 
-  MindeeClient getMindeeClient(MindeeSettings mindeeSettings) {
-    return new MindeeClient(new MindeeHttpApi(mindeeSettings));
+  @Command(name = "custom", description = "Invokes a builder API")
+  void customMethod(
+    @Parameters(
+      index = "0",
+      scope = ScopeType.INHERIT,
+      paramLabel = "<path>") File file,
+    @Parameters(
```
Comment (ianardee) [inline src/main/java/com/mindee/CLI.java:99]: should be a `"-u, --username"` required option, not a positional argument

File: src/main/java/com/mindee/CLI.java [hunk 2]
```diff
@@ -37,19 +39,6 @@ public class CLI {
     description = "Flag to set all words")
   boolean words;
 
-  @Option(names = {"-C",
```
Comment (ianardee) [inline src/main/java/com/mindee/CLI.java:40]: don't remove, but inverse the option, should be `"-c, --cut-pages"`, as by default we don't cut. When the option is activated, cut all pages after the 5th one.

File: src/main/java/com/mindee/CLI.java [hunk 3]
```diff
@@ -37,19 +39,6 @@ public class CLI {
...
-    "--no-cut-doc"},
-    scope = ScopeType.INHERIT,
-    paramLabel = "<NoCutDoc>",
-    description = "Flag to not cut a document")
-  boolean noCutDoc;
-
-  @Option(names = {"-p",
```
Comment (ianardee) [inline src/main/java/com/mindee/CLI.java:47]: agree that this should be removed

File: src/main/java/com/mindee/CLI.java [hunk 4]
```diff
@@ -37,19 +39,6 @@ public class CLI {
     description = "Flag to set all words")
   boolean words;
 
-  @Option(names = {"-C",
-    "--no-cut-doc"},
-    scope = ScopeType.INHERIT,
-    paramLabel = "<NoCutDoc>",
-    description = "Flag to not cut a document")
```
Comment (ianardee) [inline src/main/java/com/mindee/CLI.java:44]: Change text to `Keep only the first 5 pages of the document.`

File: src/main/java/com/mindee/parsing/CustomEndpointInfo.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  /**
+   * The name of the account that owns the API. Useful when using custom builder.
+   */
+  String accountName();
+
+  /**
+   * The version number of the API. Without the v (for example for the v1.2: 1.2).
+   * 1.0 by default.
+   */
+  String version() default "1.0";
```
Comment (ianardee) [inline src/main/java/com/mindee/parsing/CustomEndpointInfo.java:23]: Should be `1` by default — to always use the latest version.

File: src/main/java/com/mindee/parsing/CustomEndpointInfo.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface CustomEndpointInfo {
+  /**
+   * The name of the product associated to the expected model.
+   */
+  String endpointName();
+
+  /**
+   * The name of the account that owns the API. Useful when using custom builder.
```
Comment (ianardee) [inline src/main/java/com/mindee/parsing/CustomEndpointInfo.java:15]:
```suggestion
   * The name of the account that owns the API. Required when using custom builder.
```

**Review by ianardee — DISMISSED** [review]

File: src/main/java/com/mindee/CLI.java (hunk 1 above)
Comment (Kvindee) [reply src/main/java/com/mindee/CLI.java:99]: I do not understand, the file path is mandatory at this point

Comment (ianardee) [reply src/main/java/com/mindee/CLI.java:99]: Not sure why it's showing lines above... I put the comment on line 99 which is for `accountName`.

Path parameter does not change.

**Review by ianardee — APPROVED** [review]

**Pull request merged**